app/src/main/java/com/example/qrgo/SignupActivity.java

```diff
@@ -0,0 +1,128 @@
+ */
+public class SignupActivity extends AppCompatActivity {
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_signup);
+
+        // Hide the action bar
+        ActionBar actionBar = getSupportActionBar();
+        if (actionBar != null) {
+            actionBar.setDisplayShowTitleEnabled(false);
+            actionBar.hide();
+        }
+
+        // Hide the status bar
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+            Window window = getWindow();
+            window.setStatusBarColor(getResources().getColor(R.color.transparent));
+            window.setNavigationBarColor(getResources().getColor(R.color.transparent));
+            window.setBackgroundDrawable(new ColorDrawable(getResources().getColor(R.color.transparent)));
+        }
+
+        //Set up variables
+        final String TAG = "Sample";
+        Button mRegister;
+        final EditText username;
+        final EditText email;
+        final EditText phone;
+
+        FirebaseConnect db = new FirebaseConnect();
+
+        mRegister = findViewById(R.id.register);
+
+        // The users real name
+        username = findViewById(R.id.username);
+        email = findViewById(R.id.address);
+        phone = findViewById(R.id.phone_number);
+
+
+
+
+
+        mRegister.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                //Retrieve sign up info
+//                final String imei = getIntent().getStringExtra("imei");
+
+                final String realName = username.getText().toString();
+
+
+
+                // Traverse the string
+                for (int i = 0; i < realName.length(); i++) {
+                    if(realName.charAt(i) == ' '){
+                        final String firstName = realName.substring(0,i);
+                        final String lastName = realName.substring(i+1);
+                        final String userName = lastName.charAt(0)+firstName;
+                        final String contactEmail = email.getText().toString();
+                        final String contactPhone = phone.getText().toString();
+                        final String imei = getIntent().getStringExtra("imei");
+                        db.addNewUser(imei, userName, new FirebaseConnect.OnUserAddListener() {
+                            @Override
+                            public void onUserAdd(boolean success) {
+
+                            }
+                        });
+
+                        db.addNewPlayerProfile(userName, firstName, lastName, contactEmail, contactPhone, 0, 0, 0, new FirebaseConnect.OnUserProfileAddListener() {
+                            @Override
+                            public void onUserProfileAdd(boolean success) {
+
+                            }
+                        });
```

Comment:
Please call db.addNewPlayerProfile() inside the onUserAdd(), so that you wait for the user to be added before adding their profile.